### PR TITLE
EVG-2823: add checks for negative values in time and cost values

### PR DESCRIPTION
**Pull request by tychoish**

### CODE REVIEW

File: cloud/ec2.go [hunk 1]
```diff
@@ -769,5 +769,11 @@ func (m *ec2Manager) CostForDuration(ctx context.Context, h *host.Host, start, e
 	if err != nil {
 		return 0, errors.Wrap(err, "error fetching ebs cost")
 	}
-	return ec2Cost + ebsCost, nil
+	total := ec2Cost + ebsCost
+
+	if total < 0 {
+		return 0, errors.Errorf("cost appears to be less than 0 (%g) which is impossible", total)
```
Comment (bsamek) [inline cloud/ec2.go]: log, at debug? (and elsewhere in the diff)

File: cloud/gce_costs.go [hunk 2]
```diff
@@ -94,7 +94,7 @@ func (m *gceManager) TimeTilNextPayment(h *host.Host) time.Duration {
 // platforms (e.g. Skylake VMs), premium images (e.g. SUSE), networks, etc.
 //
 // Source: https://cloud.google.com/compute/pricing
-func (m *gceManager) CostForDuration(h *host.Host, start, end time.Time) (float64, error) {
+func (m *gceManager) bCostForDuration(h *host.Host, start, end time.Time) (float64, error) {
```
Comment (bsamek) [inline cloud/gce_costs.go]: typo

**Comment by bsamek** [comment]
We should also figure out why patch builds are not starting for your changes.

File: cloud/ec2.go (hunk 1 above)
Comment (tychoish) [reply cloud/ec2.go]: this is really low level, and will get logged in the job outcome. so we'll see it.

**Pull request merged**